packages/mongodb-chatbot-server/src/routes/conversations/addMessageToConversation.ts

```diff
@@ -249,10 +274,37 @@ async function addMessagesToDatabase({
 const loadConversation = async ({
   conversationIdString,
   conversations,
+  createConversation,
+  reqId,
+  req,
+  res,
 }: {
   conversationIdString: string;
   conversations: ConversationsService;
+  createConversation?: AddMessageToConversationRouteParams["createConversation"];
+  reqId: string;
+  req: ExpressRequest;
+  res: ExpressResponse<ApiMessage, ConversationsRouterLocals>;
 }) => {
+  // Create a new conversation if the conversationId is "null"
+  // and the route is configured to do so
+  if (
+    createConversation?.createOnNullConversationId === true &&
+    conversationIdString === "null"
+  ) {
+    logRequest({
+      reqId,
+      message: stripIndents`Creating new conversation`,
+    });
+    return await conversations.create({
+      initialMessages: createConversation.systemMessage
+        ? [createConversation.systemMessage]
+        : undefined,
+      customData: createConversation.addCustomData
+        ? await createConversation.addCustomData(req, res)
+        : undefined,
+    });
+  }
```

Comment:
(Comment is for the line below this - not included in PR)

I think we should refactor away from `toObjectId` (we only use it one time, right here) and instead inline the `ObjectId` cast. We should also use a less technical, more descriptive error message in the response.

e.g. `Invalid conversationId: ${id}`

Reply:
implemented change to inline